packages/venia-ui/lib/components/CheckoutPage/utilities.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+function isElement(element) {
+    return React.isValidElement(element);
+}
+
+const renderComponent = Comp => {
+    return isElement(Comp) ? Comp : <Comp />;
+};
+
+/**
+ * If an Else components can be React Components or Elements.
+ *
+ * React components are useful for lazy rendering.
+ */
+export const renderIf = booleanCondition => (IfComp, ElseComp) => {
```

Comment:
This is a really interesting approach to conditional rendering. Currently, throughout the app, we have code that uses memoization that returns the component. This allows us to also maintain readability in the JSX markup. Example:

```js
const maybeRenderComponent = useMemo(() -> {
  if (condition) {
    return <SomeComponent />
  }
  else {
    return null;
  }
}, [condition])

return (
  <div>
    {maybeRenderComponent}
    <SiblingComponent />
    <span>Some text</span>
  </div>
);
```

This is all to say that while I think this is a neat idea, I'm not sure we want to introduce this approach. @jimbo will likely have input here.

Reply:
@sirugh is correct. As we've discussed a few times before, in this codebase, we avoid utilities. The only utility our components use is `mergeClasses` (because we have to), but even that one is something we're aiming to eliminate.

I've worked in codebases where utility code was common. I've also used Ramda to create some beautiful functional code. Ultimately, that approach is not the right one for this project, and more importantly, it's not the one we selected in the first place. As a general rule of thumb, please write code that looks like the other code in the codebase. 😅